core/modules/shout-ability/client/shout-ability.js

```diff
@@ -1,26 +1,50 @@
+import { canUseLevelFeature } from '../../../lib/misc';
 import { moduleType } from '../../../client/helpers';
 
+
 window.addEventListener('load', () => {
-  registerModules([
-    { id: 'shout', icon: '📢', label: 'Shout', order: 40, shortcut: 55, scope: 'me' },
-    moduleType.RADIAL_MENU,
-  ]);
+  Tracker.autorun(track => {
+    if (Session.get('loading')) return;
+
+    const user = Meteor.user();
+    if (!user) return;
+
+    const isShoutFeatureEnabled = canUseLevelFeature(Meteor.user(), 'shout');
+    if (user.roles?.admin || isShoutFeatureEnabled) {
+      registerModules([
+        { id: 'shout', icon: '📢', label: 'Shout', order: 40, shortcut: 55, scope: 'me' },
+      ],
+      moduleType.RADIAL_MENU,
+      );
+    }
+
+    track.stop();
+  });
 
   hotkeys('r', { keyup: true, scope: scopes.player }, event => {
     if (event.repeat) return;
+
+    const user = Meteor.user({ fields: { _id: 1, 'profile.levelId': 1, roles: 1 } });
```

Comment:
I think we can make it simpler to avoid code repetition:
Have a variable local to the file and update it against the condition line 13. Then in the rest of the functions rely on this variable